egs2/chime7_task1/asr1/README.md

```diff
@@ -278,10 +365,71 @@ by computing SA-WER for each speaker, where the hypothesis are re-ordered based
+7. `wav-reverberate command not found` you need to install Kaldi. go to `YOUR_ESPNET_ROOT/tools/kaldi` and follow the instructions
+in `INSTALL`.
+8. `WARNING  [enhancer.py:245] Out of memory error while processing the batch` you got out-of-memory (OOM) when running GSS.
+You could try changing parameters as `gss_max_batch_dur` and in local/run_gss.sh `context-duration`
+(this latter could degrade results however). See `local/run_gss.sh` for more info. Also it could be that your GPUs are set in shared mode
+and all jobs are placed in the same GPU. You need to set them in exclusive mode.
+9. Much worse WER than baseline and you are using `run.pl`. Check the GSS results, GSS currently does not work well
+if you use multi-gpu inference and your GPUs are in shared mode. You need to run `set nvidia-smi -c 3`.
+10. `Kaldi export of Recordings with multiple audio sources is currently not supported.` you have to install lhotse from the latest
+unreleased version using `pip install git+https://github.com/lhotse-speech/lhotse`.
+
+
+
+## Memory Consumption (Useful for SLURM etc.)
+
+Figures kindly reported by Christoph Boeddeker, running this baseline code
+on Paderborn Center for Parallel Computing cluster (which uses SLURM).
+These figures could be useful to anyone that uses job schedulers and clusters
+for which resources are assigned strictly (e.g. job killed if it exceed requested
+memory resources).
+
+Used as default:
+ - train: 3G mem
+ - cuda: 4G mem (1 GPU)
+ - decode: 4G mem
+
+GSS:
+ - nj=8 (default 4)
+ - time 45 h per job (but might be faster)
+ - num_threads=5 (1 for main process, 4 for DataLoader processes)
+ - Mem: 2 GB per core or 10 GB per job. In SLURM it is --mem 2GB, since mem means mem-per-core in SLURM.
+
+scripts/audio/format_wav_scp.sh:
+ - Some spikes to the range of 15 to 17 GB
+
+`${python} -m espnet2.bin.${asr_task}_inference${inference_bin_tag`}:
+ - Few spikes to the 9 to 11 GB range.
+
+
+## Using your own Speech Separation Front-End with the pre-trained ASR model.
+
+Some suggestions from Naoyuki Kamo see https://github.com/espnet/espnet/pull/4999 <br>
+There are two possible approaches.
+
+1. After obtaining the output of the baseline GSS enhancement.
+   1. (you are more familiar with Kaldi): copy data/kaldi/{chime6,dipco,mixer6}/{dev,eval}/gss using utils/copy_data_dir.sh and then substitute the
+   file paths in each `wav.scp` manifest file with the ones produced by your approach.
+   2. (you are more familiar with lhotse): copy data/lhotse/{chime6,dipco,mixer6}/{dev,eval}/gss lhotse manifests and then
+   replace the recordings manifests with the paths to your own recordings.
+2. Directly create your Kaldi or lhotse manifests for the ASR decoding, you can follow
+either the "style" of the baseline GSS ones or the ones belonging to close-talk mics.
+
+To evaluate the new enhanced data, e.g. `data/chime6/dev/my_enhanced`, you can `run ./asr.sh --feats_type raw_copy --skip_train true --test_sets chime6/dev/enhanced`.
+
```

Comment:
Sorry, I wrote them really roughly.

- Actually we can't run asr.sh directly, because it requires train_set etc.
- I think it's better to create a wrapped script for this purpose e.g. name as `eval.sh` to run asr.sh with required options. 
- It's enough to select one from my 1 and 2 ideas.
- It's better to write an example to run the eval.sh

e.g.

```sh
utils/copy_data_dir.sh  data/kaldi/chime6/dev/gss data/kaldi/chime6/dev/my_enhanced
# rewrite wav.scp
./eval.sh --data_sets "kaldi/chime6/dev/my_enhanced"
```

I think these should be done in the next PR.

Reply:
let me delete the last bit then.

Reply:
I think diarization part  has a priority, so please just keep this comment in mind

Reply:
Yeah, I think so,  it's also okay, but it seems a little bit complicated,

Reply:
Also note that an option for `--feats_type raw_copy` is not prepared in run.sh

Reply:
Another idea is copying the dumpdir.  

```
utils/copy_data_dir.sh  dump/kaldi/chime6/dev/gss dump/kaldi/chime6/dev/my_enhanced
echo raw > dump/kaldi/chime6/dev/my_enhanced/feats_type
# Rewrite dump/kaldi/chime6/dev/my_enhanced/wav.scp
# --skip_data_prep true --skip_train true --test_sets kaldi/chime6/dev/my_enhanced
```

Reply:
So assuming that there is no ffmpeg or sox piping I could use `raw_copy` to avoid the copying right ?

Reply:
> So assuming that there is no ffmpeg or sox piping I could use raw_copy to avoid the copying right ?

Yes, that's right.